### ISSUE TITLE
(GitHub general question) GitHub issues on iOS

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#12421]
Sorry guys, I just don’t know where to start asking about GitHub web experience on iOS :D 
Please point me in the right repo (?) if you know.

Device: iPhone on iOS 17 (but also was 15, 16)
GitHub: GH in Safari as well as GH Mobile app. 

Problems are numerous, I don’t know if it ever was good at all. Experience is quite miserable to say the least :D 
1) Main issue in Safari: when I open an issue notification which has more than 10 (or so) replies, web browser usually loads only the issue page without updates first. Then I reload and either it crashes with “A problem repeatedly occurred on “FULL_LINK_INCLUDING_ISSUE_COMMENT_LINK”, or (rarely) it loads up the proper linked issue comment.
On GH Mobile, it usually loads up the top of the ticket, but with updates. I have to scroll all the way down anyway to make sure I catch new for me replies.

2) Replies with emojis can’t be viewed from GH in Safari unless I hit the same emoji (thus adding my reply) to briefly see a popup :D 

Is iOS GitHub experience as terrible for you guys? 

Any good GH clients that exists maybe?

It works absolutely great on a proper desktop though. I don’t know why they made it so bad on mobile. 

Thanks!

**Comment by Frenzie** [comment]
I don't know what it's like, but they do have an official app: https://apps.apple.com/us/app/github/id1477376905

**Comment by mergen3107** [comment]
Yep, and it is almost as terrible web experience :D that’s the one I called “GH mobile”

**Comment by Commodore64user** [comment]
I can confirm that GH on iOS sucks big time. regarding  issue two unfortunately, if you want to see who left an emoji you need to long press on the emoji (from the app, not safari). It is annoying af I know.

for issue one, safari will only point you to the last notification, again not nice. and GH mobile is just whole different mess as well and at least until yesterday's update, would not even direct you to a referenced comment, it would just throw you into the ticket leaving you wondering which comment it was referenced before (but it seems that was fixed, so yay! us).

On iPhone, I just have the mobile app and a web app (I don't load it from safari directly anymore), and will switch between them according to my needs at the time. For example if I need to find (or see) some code, web app, if it is a quick check to read a comment, mobile app.

the web app also won't let you refresh the content easily, there is another fun thing to think about.

in short, It sucks! there is no other way to put it.

Pro tip: add a bunch of text replacement entries on the iOS keyboard settings for things you might need that are annoying to type on the fly. for example, if I type`"" "` three of those together (without the space) it will replace them for the much harder to type ` `` ` or `<> k` (again without the space) will replace it with `<kbd></kbd>` and if you have a Mac they will work there too so yay!

**Comment by mergen3107** [comment]
@Commodore64user 
Thank you very much for your valuable inputs!

Glad to know they fixed it in the app (I’ll go reinstall lol)

Yep, shortcuts for the win!
I replace `qqq` with ``` because this one is the hardest to type on keyboard :D I chose “q” letter because it is the closest one to the actual ` key on PC :D

**Comment by mergen3107** [comment]
What does your “<> k” shortcut do by the way?

**Comment by Commodore64user** [comment]
> What does your “<> k” shortcut do by the way?

Say i want to tell you to use shortcut alt+k then i would type "<> kk" without the space and that would populate <kbd></kbd> + <kbd></kbd>

**Comment by Commodore64user** [comment]
Ah fuck, they just ruined commenting from the mobile app. Before yesterday's update, you could minimise a comment to copy or re-read something and you could get back to where you left off later. Now if you try to minimise it, it will ask you if you want to discard it, and if you say no, it just keeps it open, why??? I guess comments now need to move to the web app.

It's as if the people working on it don't use it themselves.

**Comment by mergen3107** [comment]
Most of the iOS GitHub app issues are gone, including saving the draft response.
Version 1.181.0
